pallets/xcmp-handler/src/lib.rs

```diff
@@ -107,6 +111,9 @@ pub mod pallet {
 
 		/// Utility for determining XCM instruction weights.
 		type Weigher: WeightBounds<<Self as pallet::Config>::Call>;
+
+		/// XCM Flow Selection Matcher
+		type XcmFlowSelector: XcmFlowSelector;
```

Comment:
The idea with this design is that the runtime will pass in a matcher that decides which `ParaId` requires which xcm flow based on what the partner decides they want. This needs to be passed down from the runtime since the same `ParaId` might use two different flows in different envs.

Reply:
Eventually we can have this be updatable by governance. In this alpha design it's set by the runtime developer.